### PR TITLE
Decorator

### PR DESCRIPTION
**Pull request by shaqayq**
In this part of the project, I used the Decorator design pattern to correct the names of people and added `CapitalizeDecorator `, `TrimmerDecorator`, `nameable `, and `Decorator `classes.

### CODE REVIEW

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require_relative 'base_decorator'
+class TrimmerDecorator < Decorator
+  def correct_nam
+    @nameable.correct_name.length > 10 ? @nameable.correct_nam[0..9] : @nameable.correct_name
```
Comment (Davidosky007) [inline trimmer_decorator.rb]: * Kindly fix the TrimerDecorator class as it doesn't work as expected. HINT 💭 your correct_name method is spelled wrongly as shown in my screenshot below kindly fix it as when i did i got the expected result 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @shaqayq 👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Well written commit messages 👍 
* Proper use of Gitflow 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 1 above)
Comment (shaqayq) [reply trimmer_decorator.rb]: @Davidosky007!
Thank you for reviewing my code. I will change it.

**Review by cwaku — APPROVED** [review]
Hi @shaqayq 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
 - [x] Correct GitHub Flow 👍🏾 
 - [x] Good commit history 🥇 
 - [x] Good PR title/description 💯 
 - [x] All linters passed 🥇 
 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. The previous reviewer simply missed some of them, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**